cl/search/tests/test_pacer_bulk_fetch.py

```diff
@@ -0,0 +1,640 @@
+from unittest.mock import MagicMock, patch
+
+from django.core.cache import cache as django_cache
+from django.core.management import call_command
+from django.utils import timezone
+
+from cl.recap.factories import PacerFetchQueueFactory
+from cl.recap.models import PROCESSING_STATUS, PacerFetchQueue
+from cl.search.factories import (
+    CourtFactory,
+    DocketEntryFactory,
+    DocketFactory,
+    RECAPDocumentFactory,
+)
+from cl.search.management.commands.pacer_bulk_fetch import (
+    Command,
+    append_value_in_cache,
+)
+from cl.search.models import RECAPDocument
+from cl.tests.cases import TestCase
+from cl.users.factories import UserFactory
+
+
+def _clean_cache_keys(keys):
+    for key in keys:
+        django_cache.delete(key)
+
+
+class BulkFetchPacerDocsTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = UserFactory()
+        cls.courts = []
+        for i in range(5):
+            court = CourtFactory(id=f"ca{i}", jurisdiction="F")
+            cls.courts.append(court)
+
+        cls.dockets = []
+        for court in cls.courts:
+            docket = DocketFactory(court=court)
+            cls.dockets.append(docket)
+
+        # Docket entries by court index
+        cls.docket_entries = {
+            0: [DocketEntryFactory(docket=cls.dockets[0]) for _ in range(300)],
+            1: [DocketEntryFactory(docket=cls.dockets[1]) for _ in range(8)],
+            2: [DocketEntryFactory(docket=cls.dockets[2]) for _ in range(20)],
+            3: [DocketEntryFactory(docket=cls.dockets[3]) for _ in range(15)],
+            4: [DocketEntryFactory(docket=cls.dockets[4]) for _ in range(10)],
```

Comment:
Is it possible to use fewer factories for these tests? I think we might be able to assert the same behavior with just a few of them? for example, instead of creating 300 for DocketEntry 0, we could create just 10, and so on.

Reply:
Reduced the number of factories.